src/pynwb/ndx_nirs/__init__.py

```diff
@@ -119,15 +124,39 @@ def __init__(self, **kwargs):
     },
     *update_docval(
         DynamicTable.__init__,
-        name={"default": "channels"},
-        description={"default": "A table describing optical channels of a NIRS device"},
+        updates=dict(
+            name={"default": "channels"},
+            description={
+                "default": "A table describing optical channels of a NIRS device"
+            },
+        ),
     ),
 ]
 
 
 @register_class("NIRSChannelsTable", "ndx-nirs")
 class NIRSChannelsTable(DynamicTable):
-    """A DynamicTable representing the optical channels of this NIRS device"""
+    """A DynamicTable representing the optical channels of this NIRS device.
+
+    Before adding channels to the table, the sources and detectors for the
+    device need to be defined in a NIRSSourcesTable and a NIRSDetectorsTable,
+    respectively. These can be passed in either during class initializing or
+    afterwards.
+
+    Examples:
+    ```
```

Comment:
I am not sure what doc generation is being used, but do you need to specify python for highlighting?
Regardless, I love a good example in the docs, very handy.

Reply:
Good point. We don't have doc generation working yet, that's on our backlog, so I'm not sure yet. The extension template included some sphinx integration, so I imagine we'd go with that, but we haven't gotten it working yet (or decided on a docstring style). I'll mark the block as python for now, and we'll need to check all docstrings once we get the doc generation working.